lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -9,18 +9,16 @@
 	/**
 	 * Whether a theme or its parent have a theme.json file.
 	 *
-	 * @param boolean $clear_cache Whether the cache should be cleared and theme support recomputed. Default is false.
+	 * The result would be cached via the WP_Object_Cache
+	 * under the `wp_theme_has_theme_json` key.
 	 *
 	 * @return boolean
 	 */
-	function wp_theme_has_theme_json( $clear_cache = false ) {
-		static $theme_has_support = null;
-
-		if ( true === $clear_cache ) {
-			$theme_has_support = null;
-		}
-
-		if ( null !== $theme_has_support ) {
+	function wp_theme_has_theme_json() {
+		$cache_key         = 'wp_theme_has_theme_json';
+		$cache_found       = false;
+		$theme_has_support = wp_cache_get( $cache_key, '', false, $cache_found );
```

Comment:
I had to check the code reference to see what `$found` does. Unfortunately, it's not apparent that the parameter is passed by reference.

I've not seen it used in the core much, but this might have changed.

My personal preference would be to avoid using it.

Reply:
It is used in core [here](https://github.com/WordPress/wordpress-develop/blob/50ebc187ab09b035595077dce29fc28825a82b4c/src/wp-includes/class-wp-query.php#L3114).

Reply:
Thanks for the reference, @spacedmonkey.

Do you think we need to use the `$found` argument in this function?

Reply:
Yeah we do. The default return of `wp_cache_get` is false if not found. We are storing a boolean value. So we need to use.

Reply:
Why don't we just store `0` / `1` instead? I'm also a bit wary of using the `$found` parameter since, while it's part of core, is not at all commonly used to the degree that I'm afraid some cache implementations don't implement it correctly. I think not storing a boolean here and fixing it that way would be a safer call IMO.

Reply:
```suggestion
		$theme_has_support = wp_cache_get( $cache_key, 'theme', false );
```

Reply:
Added group and remove cache found.

Reply:
I'm going to store ints. I considered using true/false as strings, though I see how it'll be tempting for our future selves to just convert them to boolean and ignore the rationale for this. I also added a comment to make sure the rationale is documented.